src/Tasks/Microsoft.NET.Build.Tasks/PublishAssembliesResolver.cs

```diff
@@ -31,6 +31,11 @@ public IEnumerable<ResolvedFile> Resolve(ProjectContext projectContext)
 
             foreach (LockFileTargetLibrary targetLibrary in projectContext.GetRuntimeLibraries(_privateAssetPackageIds))
             {
+                if (targetLibrary.Type == "project")
```

Comment:
Should this only read ``package`` type libraries instead? It might be safer in the scenario where a new type shows up. Not sure on which better matches the current build task's approach, just wanted to point it out.